src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -496,6 +496,8 @@ BoundExpression makeConditionalAccess(BoundExpression receiver)
 
             static bool isFieldOfMarshalByRef(BoundExpression expr, CSharpCompilation compilation)
             {
+                Debug.Assert(!IsCapturedPrimaryConstructorParameter(expr));
```

Comment:
>Debug.Assert(!IsCapturedPrimaryConstructorParameter(expr)) [](http://example.com/codeflow?start=16&length=58)

How do we know the expression is not a primary constructor parameter?

Reply:
> How do we know the expression is not a primary constructor parameter?

The ```expr``` is supposed to be lowered before we get here. Captured parameters are replaced with fields during lowering.